cmd/metrics.go

```diff
@@ -81,117 +83,153 @@ func (c *minioCollector) Collect(ch chan<- prometheus.Metric) {
+	if !exists {
+		return
+	}
+	healMetricsNamespace := "self_heal"
 	ch <- prometheus.MustNewConstMetric(
 		prometheus.NewDesc(
-			prometheus.BuildFQName("minio", "disks", "total"),
-			"Total number of disks for current MinIO server instance",
+			prometheus.BuildFQName(healMetricsNamespace, "time", "since_last_activity"),
+			"Time elapsed (in nano seconds) since last self healing activity",
 			nil, nil),
 		prometheus.GaugeValue,
-		float64(totalDisks.Sum()),
+		float64(time.Since(bgSeq.lastHealActivity)),
 	)
-
-	for i := 0; i < len(storageInfo.Total); i++ {
-		mountPath, total, free := storageInfo.MountPaths[i], storageInfo.Total[i],
-			storageInfo.Available[i]
-
-		// Total disk usage by the disk
+	for k, v := range bgSeq.getScannedItemsMap() {
 		ch <- prometheus.MustNewConstMetric(
 			prometheus.NewDesc(
-				prometheus.BuildFQName("disk", "storage", "used"),
-				"Total disk storage used on the disk",
-				[]string{"disk"}, nil),
+				prometheus.BuildFQName(healMetricsNamespace, "objects", "scanned"),
+				"Objects scanned in current self healing run",
+				[]string{"type"}, nil),
 			prometheus.GaugeValue,
-			float64(total-free),
-			mountPath,
+			float64(v), string(k),
 		)
-
-		// Total available space in the disk
+	}
+	for k, v := range bgSeq.getHealedItemsMap() {
 		ch <- prometheus.MustNewConstMetric(
 			prometheus.NewDesc(
-				prometheus.BuildFQName("disk", "storage", "available"),
-				"Total available space left on the disk",
-				[]string{"disk"}, nil),
+				prometheus.BuildFQName(healMetricsNamespace, "objects", "healed"),
+				"Objects healed in current self healing run",
+				[]string{"type"}, nil),
 			prometheus.GaugeValue,
-			float64(free),
-			mountPath,
+			float64(v), string(k),
 		)
-
-		// Total storage space of the disk
+	}
+	for k, v := range bgSeq.gethealFailedItemsMap() {
+		// healFailedItemsMap stores the endpoint and volume state separated by comma,
+		// split the fields and pass to channel at correct index
+		s := strings.Split(k, ",")
 		ch <- prometheus.MustNewConstMetric(
 			prometheus.NewDesc(
-				prometheus.BuildFQName("disk", "storage", "total"),
-				"Total space on the disk",
-				[]string{"disk"}, nil),
+				prometheus.BuildFQName(healMetricsNamespace, "objects", "heal_failed"),
+				"Objects for which healing failed in current self healing run",
+				[]string{"mount_path", "volume_status"}, nil),
 			prometheus.GaugeValue,
-			float64(total),
-			mountPath,
+			float64(v), string(s[0]), string(s[1]),
 		)
 	}
+}
 
-	connStats := globalConnStats.toServerConnStats()
-
-	// Network Sent/Received Bytes (internode)
+// collects gateway specific metrics for MinIO instance in Prometheus specific format
+// and sends to given channel
+func gatewayMetricsPrometheus(ch chan<- prometheus.Metric) {
+	if !globalIsGateway || (globalGatewayName != "s3" && globalGatewayName != "azure" && globalGatewayName != "gcs") {
+		return
+	}
+	m, _ := globalObjectAPI.GetMetrics(context.Background())
```

Comment:
```suggestion
	m, _ := globalObjectAPI.GetMetrics(context.Background())
```

Using `globalObjectAPI`  directly is not allowed and shouldn't be used, I see that the current code used it improperly.  Please use `newObjectLayerWithoutSafeModeFn().GetMetrics`

Reply:
let's not ignore an error because - `m` is a pointer and it may be `nil`